INSTALL.md

```diff
@@ -1,96 +1,190 @@
+- [omniorb](http://omniorb.sourceforge.net), 4.2.1 or newer
+- [libzmq](https://github.com/zeromq/libzmq), 4.0.5 or newer
+- [cppzmq](https://github.com/zeromq/cppzmq), 4.2.2 or newer
 
-- `git clone https://github.com/tango-controls/cppTango`
-- `cd cppTango`
-- `mkdir build`
-- `cd build`
-- `cmake ..`
-- `make [-j NUMBER_OF_CPUS]`
-- `sudo make install`
-
-## Available are the following cmake options:
-
-- `-DCMAKE_INSTALL_PREFIX=<desired installation path>`
-- `-DIDL_BASE=<tango-idl installation folder>`
-- `-DOMNI_BASE=<omniORB4 home folder>`
-- `-DCPPZMQ_BASE=<cppzmq home folder>`
-- `-DZMQ_BASE=<zmq home folder>`
-- `-DCMAKE_BUILD_TYPE=<Release|Debug>` (`Release` by default)
-- `-DCMAKE_VERBOSE_MAKEFILE=<ON|OFF>`
-- `-DTANGO_USE_USING_NAMESPACE=<ON|OFF>` choose `OFF` for modern builds
-- `-DUSE_PCH=<ON|OFF>` Use precompiled headers (`OFF` by default)
-- `-DBUILD_TESTING=<ON|OFF>` Build the test suite (`ON` by default)
-- `-DBUILD_SHARED_LIBS=<ON|OFF>` Build tango as shared library (`ON` by default)
-- `-DWARNINGS_AS_ERRORS=<ON|OFF>` Treat compiler warnings as errors (`OFF` by default)
-- `-DTANGO_JPEG_MMX=<ON|OFF>` Build the jpeg support using MMX extensions (`ON` by default)
-- `-DCOVERALLS=<ON|OFF>` Generate code coverage data (`OFF` by default)
-- `-DTANGO_INSTALL_DEPENDENCIES=<ON|OFF>` Install dependencies of tango as well (`ON` by default, windows only)
-
-Typical output:
+In the following we assume a linux-based system, see [here](#building-on-windows) for building on Windows.
+
+On current debian systems the dependencies, except tango-idl, are available as distribution packages:
 
+```bash
+sudo apt install cmake build-essential git libcos4-dev libomniorb4-dev libomnithread4-dev libzmq3-dev omniidl python3
 ```
--- Install configuration: "Debug"
--- Installing: /storage/Projects/org.tango/git/cppTango/build/install/lib/libtango.so.9.2.5
--- Installing: /storage/Projects/org.tango/git/cppTango/build/install/lib/libtango.so.9
--- Installing: /storage/Projects/org.tango/git/cppTango/build/install/lib/libtango.so
--- Installing: /storage/Projects/org.tango/git/cppTango/build/install/lib/pkgconfig/tango.pc
--- Installing: /storage/Projects/org.tango/git/cppTango/build/install/include/log4tango/Appender.hh
-<snip>
--- Installing: /storage/Projects/org.tango/git/cppTango/build/install/include/subdev_diag.h
--- Installing: /storage/Projects/org.tango/git/cppTango/build/install/include/encoded_attribute.h
--- Installing: /storage/Projects/org.tango/git/cppTango/build/install/include/encoded_format.h
--- Installing: /storage/Projects/org.tango/git/cppTango/build/install/include/idl/tango.h
+
+If your linux does not have precompiled packages for these dependencies jump to the
+[next](#compiling-the-dependencies) section for compilation instructions.
+
+## Compiling tango
+
+## tango-idl
+
+```bash
+git clone https://github.com/tango-controls/tango-idl
+cd tango-idl
+mkdir build
+cd build
+cmake ..
+make [-j NUMBER_OF_CPUS]
```

Comment:
You don't need this step, 'all' target is empty in tango-idl, and anyway '-j' won't speed up anything in such simple package.

BTW I noticed that you can't run these steps without c++ compiler being installed, which makes no sense. But maybe this requirement comes from CMake.

Reply:
Good point. Done.